src/TimerDialog.cpp

```diff
@@ -0,0 +1,250 @@
+    unsigned int minutes = 0;
+    unsigned int hours = 0;
+   if (t == 0)
+       return "00:00:00";
+   if (t > 59){
+       seconds = t%60;
+       minutes = t/60;
+       if (minutes > 59){
+          hours = minutes/60;
+          minutes = minutes%60;
+       }
+   }
+   QString secStr, minStr, hourStr;
+   if (seconds < 10)
+       secStr = "0" + QString::number(seconds);
+   else
+       secStr = QString::number(seconds);
+   if (minutes <10)
+       minStr = "0" + QString::number(minutes);
+   else
+       minStr = QString::number(minutes);
+   if (hours < 10)
+       hourStr = "0" + QString::number(hours);
+   else
+       hourStr = QString::number(hours);
+   return hourStr + ":" + minStr + ":" + secStr;
+}
+
+void TimerDialog::decrementTime()
+{
+    if (started) {
+        if (stopped)
+            flash();
+        else if (time == 0)
+            timesUp();
+        else {
+            time = time - 1;
+            updateTime();
+        }
+    }
+}
+
+void TimerDialog::on_setSoundButton_clicked()
+{
+    //Taken directly form old timers
+    QDir soundsDir = QString("%1sounds/").arg(Brewtarget::getDataDir());
+    QString soundFile = QFileDialog::getOpenFileName( qobject_cast<QWidget*>(this), tr("Open Sound"), soundsDir.exists() ? soundsDir.canonicalPath() : "", tr("Audio Files (*.wav *.ogg *.mp3 *.aiff)") );
+
+    if( soundFile.isNull() )
+    {
+       Brewtarget::logW("Null sound file.");
+       return;
+    }
+    setSound(soundFile);
+    // Indicate a sound is loaded
+    setSoundButton->setCheckable(true);
+    setSoundButton->setChecked(true);
+}
+
+void TimerDialog::setDefualtAlarmSound()
+{
+    QString soundFile = QString("%1sounds/").arg(Brewtarget::getDataDir()) + "beep.ogg";
+    setSound(soundFile);
+}
+
+void TimerDialog::setSound(QString s)
+{
+#ifndef NO_QTMULTIMEDIA
+   if( !playlist->clear() )
+      Brewtarget::logW(playlist->errorString());
+   if( !playlist->addMedia(QUrl::fromLocalFile(s)) )
+      Brewtarget::logW(playlist->errorString());
+   playlist->setCurrentIndex(0);
+#endif
+}
+
+void TimerDialog::on_setTimeBox_valueChanged(int t)
+{
+    if (t*60 > boilTime->getTime()) {
+        QMessageBox::warning(this, "Error", "Addition time cannot be longer than remaining boil time");
+        time = 0;
+        started = false;
+    } else {
+        setTime(t*60);
+    }
+}
+
+void TimerDialog::timesUp()
+{
+    if (!stopped) {
+        if (this->isHidden())
+            this->show();
+        this->setFocus();
+        this->raise();
+        startAlarm();
+        stopped = true;
+    }
+    else
+        flash();
+}
+
+void TimerDialog::flash()
+{
+    oldColors = ! oldColors;
+
+    if( oldColors )
+       timeLCD->setPalette(paletteOld);
+    else
+       timeLCD->setPalette(paletteNew);
+
+    timeLCD->repaint();
+}
+
+void TimerDialog::reset()
+{
+    time = boilTime->getTime() - (setTimeBox->value() * 60);
+    if (stopped)
+        stopped = false;
+    mediaPlayer->stop();
```

Comment:
Aah cool, I'll sort that out. I took that from your code.
On 06 Dec 2015 19:56, "Philip Lee" <notifications@github.com> wrote:

> In src/TimerDialog.cpp
> <https://github.com/Brewtarget/brewtarget/pull/122#discussion_r46773467>:
>
> > +    oldColors = ! oldColors;
> > +
> > +    if( oldColors )
> > +       timeLCD->setPalette(paletteOld);
> > +    else
> > +       timeLCD->setPalette(paletteNew);
> > +
> > +    timeLCD->repaint();
> > +}
> > +
> > +void TimerDialog::reset()
> > +{
> > +    time = boilTime->getTime() - (setTimeBox->value() * 60);
> > +    if (stopped)
> > +        stopped = false;
> > +    mediaPlayer->stop();
>
> Not wrapped in #ifndef NO_QTMULTIMEDIA causes compiler error.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Brewtarget/brewtarget/pull/122/files#r46773467>.
>